test/sets.jl

```diff
@@ -252,3 +252,13 @@ end
 @test pop!(Set(1:2), 2, nothing) == 2
 
 @test length(Set(['x',120])) == 2
+
+# convert
+let
+    iset = Set([17, 4711])
+    @test convert(Set{Float64}, iset) == iset
+    fset = Set([17.0, 4711.0])
+    @test convert(Set{Int}, fset) == fset
```

Comment:
In a hurry this morning, so I only dug through the manual briefly and pushed the cleanest (still dirty) way I could think of doing this.  More julian suggestions are most welcome.